controllers/webhook/status_updater.go

```diff
@@ -0,0 +1,183 @@
+)
+
+func completionStateForRunCompletionStatus(rcs common.RunCompletionStatus) *pipelinesv1.CompletionState {
+	switch rcs {
+	case common.RunCompletionStatuses.Succeeded:
+		return &pipelinesv1.CompletionStates.Succeeded
+	case common.RunCompletionStatuses.Failed:
+		return &pipelinesv1.CompletionStates.Failed
+	default:
+		return nil
+	}
+}
+
+type StatusUpdater struct {
+	K8sClient client.Client
+}
+
+func NewStatusUpdater(ctx context.Context, scheme *runtime.Scheme) (StatusUpdater, error) {
+	logger := log.FromContext(ctx)
+	k8sConfig, err := common.K8sClientConfig()
+	if err != nil {
+		logger.Error(err, "Error reading k8s client config")
+		return StatusUpdater{}, err
+	}
+
+	k8sClient, err := client.New(k8sConfig, client.Options{
+		Scheme: scheme,
+	})
+	if err != nil {
+		logger.Error(err, "Error creating k8s client")
+		return StatusUpdater{}, err
+	}
+	logger.Info("StatusUpdater created")
+	return StatusUpdater{k8sClient}, nil
+}
+
+func (su StatusUpdater) handle(
+	ctx context.Context,
+	event common.RunCompletionEvent,
+) error {
+	if event.RunName != nil {
+		if err := su.completeRun(ctx, event); err != nil {
+			return err
+		}
+	}
+	if event.RunConfigurationName != nil {
+		if err := su.completeRunConfiguration(ctx, event); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (su StatusUpdater) completeRun(ctx context.Context, event common.RunCompletionEvent) error {
+	logger := log.FromContext(ctx)
+
+	if event.RunName.Namespace == "" {
+		logger.Info(
+			"RunCompletionEvent's RunName namespace was the empty string. Skipping.",
+			"RunId",
+			event.RunId,
+		)
+		return nil
+	}
+
+	run := pipelinesv1.Run{}
+
+	if err := su.K8sClient.Get(
+		ctx,
+		types.NamespacedName{
+			Namespace: event.RunName.Namespace,
+			Name:      event.RunName.Name,
+		},
+		&run,
+	); err != nil {
+		if errors.IsNotFound(err) {
+			logger.Info(
+				"RunCompletionEvent's Run was not found. Skipping.",
+				"RunId",
+				event.RunId,
+				"RunName",
+				event.RunName,
+				"Action",
+				"Get",
+			)
+			return nil
+		}
+		return err
+	}
+
+	if completionState := completionStateForRunCompletionStatus(event.Status); completionState != nil {
+		run.Status.CompletionState = *completionState
+		if err := su.K8sClient.Status().Update(ctx, &run); err != nil {
+			if errors.IsNotFound(err) {
+				logger.Info(
+					"RunCompletionEvent's Run was not found. Skipping.",
+					"RunId",
+					event.RunId,
+					"RunName",
+					event.RunName,
+					"Action",
+					"Update",
+				)
+				return nil
+			}
+			return err
+		}
+	}
+	return nil
+}
+
+func (su StatusUpdater) completeRunConfiguration(
+	ctx context.Context,
+	event common.RunCompletionEvent,
+) error {
+	logger := log.FromContext(ctx)
+
+	if event.Status != common.RunCompletionStatuses.Succeeded ||
+		event.RunConfigurationName.Namespace == "" {
+		logger.Info(
+			"RunCompletionEvent's RunConfigurationName namespace was the empty string. Skipping.",
```

Comment:
```suggestion
			"RunCompletionEvent's RunConfigurationName namespace in empty. Skipping.",
```

Reply:
I don't think so, because `RunConfigurationName` is a ptr, therefore "empty" could mean nil, whereas empty string means `RunConfigurationName` was not nil, but the `Namespace` was the empty string.

Reply:
Its clear  the `namespace` was empty and not the `RunConfigurationName` so I don't think the pointer issue is relevant in this case

Reply:
https://github.com/sky-uk/kfp-operator/pull/393/commits/d15e740eba89062895ea9f4aabc7d27ed6515859